crates/crypto/src/key_manager/key_store/in_memory_key_store.rs

```diff
@@ -0,0 +1,71 @@
+use crate::key::PrivateKey;
+use crate::key_manager::key_store::{KeyStore, KeyStoreError};
+use std::collections::HashMap;
+use std::sync::Mutex;
+
+/// An in-memory implementation of the [`KeyStore`] trait.
+pub struct InMemoryKeyStore {
+    map: Mutex<HashMap<String, PrivateKey>>,
+}
```

Comment:
not very familiar w/ rust.. why is `Mutex` used here over something like `RwLock`?

Reply:
This is a really good question.

* `Mutex` allows mutually exclusive access to the shared data, no matter if the caller is accessing the data for a read or write. Both read and writes will block the current thread until it acquires the lock. This is why there's only a singular `lock()` function to acquire a lock into the underlying data.
* `RwLock` allows for any number of readers, but requires exclusive access for writes. This means that a read lock will only block the current thread if there's an active write lock. This is why there's distinctive `read()` and `write()` functions to acquire a lock to the underlying data.

This makes `RwLock` preferable in read-heavy situations, where as `Mutex` is more appropriate for write-heavy situations.

In our case, writing keys to the store is going to be FAR less frequent than reading them. So we actually _should_ be using a `RwLock` here for better performance. I'll go ahead and update!

Thank you for asking this! Learning together!

Docs:
* [Mutex](https://doc.rust-lang.org/std/sync/struct.Mutex.html)
* [RwLock](https://doc.rust-lang.org/std/sync/struct.RwLock.html)